### ISSUE TITLE
Except for the datetime fields, The filter in the dashboard cannot be connected to the nested SQL question

### ISSUE THREAD

**Issue by 8debug** [metabase/metabase#15537]
**Describe the bug**
Except for the datetime fields, The filter in the dashboard cannot be connected to the nested SQL question

**Logs**

**To Reproduce**
1. Native query > My Oracel Dataset > Write complex SQL statements
note: the execution results of the statement include datetime type fields, string type fields, and number type fields

2. Save question as Q1
3. Simple question > Saved Questions > Q1
4. Save question as Q2 and add to dashboard
5. Add two filter linked to Q2, a time "All options" and a "Other Categories" and save dashboard

**Expected behavior**
"Time" filter can connect to Q2 datetime type fields;
"Other Categories" filter can connect to Q2 string type fields;
note: Sorry, The tutorial for using the ID filter was not found

**Information about your Metabase Installation:**
- Chrome 88.0.4324.104
- Windows 10 Pro
- Oracel 11g
- Metabase version: v0.38.2、v0.38.3
- Metabase hosting environment: Windows Server 2016 Datacenter、Centos7
- Metabase internal database: H2

    
![image](https://user-images.githubusercontent.com/3394927/114121419-e54d8c80-9920-11eb-8195-cac812279e1a.png)

**Comment by flamber** [comment]
Hi @8debug 
Looks like a duplicate of #13186.
I cannot reproduce. Can you post the query you're using in Q1?
And please post the content of `report_card.result_metadata` for Q1 and Q2, since I'm quite sure the problem is there.

**Comment by 8debug** [comment]
> Hi @8debug
> Looks like a duplicate of #13186.
> I cannot reproduce. Can you post the query you're using in Q1?
> And please post the content of `report_card.result_metadata` for Q1 and Q2, since I'm quite sure the problem is there.

Can you tell me the steps to get `report_card.result_metadata`? I don’t know how to operate it.

**Comment by flamber** [comment]
@8debug That is in your application database. If you're using H2, then you should migrate away from that if you're using Metabase in production: https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html

**Comment by 8debug** [comment]
> @8debug That is in your application database. If you're using H2, then you should migrate away from that if you're using Metabase in production: https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html

I still don't understand what you mean. Is it necessary for me to provide a complete SQL statement? SQL is very complicated, it is Oracle

**Comment by 8debug** [comment]
> @8debug That is in your application database. If you're using H2, then you should migrate away from that if you're using Metabase in production: https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html

Use the following SQL statement and follow the above steps to reproduce my problem. 
 **note：The date filter on the dashboard cannot be connected to the problem too.**

```sql
select 
    o.TOTAL 订单总金额, 
    o.CREATED_AT 订单时间, 
    p.CITY 顾客所在城市, 
    p.SOURCE 顾客来源, 
    s.TITLE 商品名, 
    s.CATEGORY 商品类型, 
    s.VENDOR 生产厂商, 
    s.RATING 评分
from ORDERS o, PEOPLE p, PRODUCTS s 
where o.USER_ID = p.ID and o.PRODUCT_ID = s.ID
```

**Comment by 8debug** [comment]
I found a serious problem,  I will save the following four SQL statements as Q1-1, Q1-2, Q1-3, Q1-4,

### Q1-1 ( No column alias )
``` 
select 
    o.TOTAL, 
    o.CREATED_AT, 
    p.CITY, 
    p.SOURCE, 
    s.TITLE, 
    s.CATEGORY, 
    s.VENDOR, 
    s.RATING 

from ORDERS o, PEOPLE p, PRODUCTS s 
where o.USER_ID = p.ID and o.PRODUCT_ID = s.ID
```

### Q1-2 ( All column aliases are Chinese characters )
``` 
select 
    o.TOTAL 订单总金额, 
    o.CREATED_AT 订单时间, 
    p.CITY 顾客所在城市, 
    p.SOURCE 顾客来源, 
    s.TITLE 商品名, 
    s.CATEGORY 商品类型, 
    s.VENDOR 生产厂商, 
    s.RATING 评分
from ORDERS o, PEOPLE p, PRODUCTS s 
where o.USER_ID = p.ID and o.PRODUCT_ID = s.ID
```

### Q1-3 ( The column alias part is Chinese characters )
```
select 
    o.TOTAL, 
    o.CREATED_AT my_时间, 
    p.CITY my_城市, 
    p.SOURCE my_来源, 
    s.TITLE my_title, 
    s.CATEGORY my_分类, 
    s.VENDOR my_厂家, 
    s.RATING 
from ORDERS o, PEOPLE p, PRODUCTS s 
where o.USER_ID = p.ID and o.PRODUCT_ID = s.ID
```

### Q1-4 ( All column aliases are in English )
```
select 
    o.TOTAL my_total, 
    o.CREATED_AT my_create, 
    p.CITY my_city, 
    p.SOURCE my_source, 
    s.TITLE my_tile, 
    s.CATEGORY my_category, 
    s.VENDOR my_vendor, 
    s.RATING my_rating

from ORDERS o, PEOPLE p, PRODUCTS s 
where o.USER_ID = p.ID and o.PRODUCT_ID = s.ID
```

Create four simple questions separately and quote Q1-1, Q1-2, Q1-3, Q1-4 respectively.

Sum of TOTAL, by CREATED_AT:Year, SOURCE

Then save them separately as Q2-1, Q2-2, Q2-3, Q2-4, and display them in the same dashboard

Add a category filter to the dashboard and find that the category filter cannot connect all questions, and the connection items for each question are different

This should be a bug, and Which fields are displayed as the connection items of the category filter according to which rules of the question?

**Comment by flamber** [comment]
@8debug I cannot reproduce on 0.38.3 - only some columns, which isn't available as noted in https://github.com/metabase/metabase/issues/13186#issuecomment-683830860

If you can reproduce this, then:
1. Post "Diagnostic Info" from Admin > Troubleshooting
2. Post the content from the application database table `report_card.result_metadata` for both the SQL and Saved Question
3. And just include 1 example - there's not reason to provide 4 different examples if they all fail.

**Comment by 8debug** [comment]
> @8debug I cannot reproduce on 0.38.3 - only some columns, which isn't available as noted in [#13186 (comment)](https://github.com/metabase/metabase/issues/13186#issuecomment-683830860)
> ![image](https://user-images.githubusercontent.com/1447303/114466237-802bbc80-9be8-11eb-9bf0-5d0014114ea4.png)
> 
> If you can reproduce this, then:
> 
> 1. Post "Diagnostic Info" from Admin > Troubleshooting
> 2. Post the content from the application database table `report_card.result_metadata` for both the SQL and Saved Question
> 3. And just include 1 example - there's not reason to provide 4 different examples if they all fail.

Your steps are wrong, You can first follow my steps to put Q2-1, Q2-2, Q2-3, Q2-4 in the same dashboard, and then add a category filter to compare and view bugs. I think I need to create a new issue
